core/lib/contract_verifier/src/etherscan/solc_versions_fetcher.rs

```diff
@@ -0,0 +1,208 @@
+use std::{
+    collections::HashMap,
+    sync::Arc,
+    time::{Duration, Instant},
+};
+
+use reqwest::Client;
+use serde::Deserialize;
+use tokio::sync::Mutex;
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+struct Build {
+    version: String,
+    long_version: String,
+    prerelease: Option<String>,
+}
+
+#[derive(Deserialize, Debug)]
+struct Response {
+    builds: Vec<Build>,
+}
+
+const SOLC_BUILDS_URL: &str =
+    "https://raw.githubusercontent.com/ethereum/solc-bin/refs/heads/gh-pages/bin/list.json";
+
+/// Fetches solc long versions from the official solc-bin repository.
+/// Long build versions are required for verifying contracts on Etherscan.
+#[derive(Debug, Clone)]
+pub(super) struct SolcVersionsFetcher {
+    http_client: Client,
+    solc_long_versions: HashMap<String, String>,
+    last_update: Arc<Mutex<Option<Instant>>>,
+}
+
+impl Default for SolcVersionsFetcher {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl SolcVersionsFetcher {
+    /// Creates a new instance of SolcVersionsFetcher. Populates the default list of solc versions.
+    pub fn new() -> Self {
+        let client = Client::builder()
+            // Timeout for establishing connection
+            .connect_timeout(Self::connect_timeout())
+            // Total timeout for the request
+            .timeout(Self::request_timeout())
+            .build()
+            .expect("Failed to create a reqwest client for SolcVersionsFetcher");
+
+        // This is a hardcoded list of solc versions that are supported by Etherscan
+        let mut solc_long_versions = std::collections::HashMap::new();
```

Comment:
To make it a bit more readable, you can define a constant outside of this function, and then create a hashmap from iter, e.g. (pseudocode):

```rust
/// Statically known version mapping
const SOLC_VERSIONS: &[(&str, &str)] = &[
  ("0.4.11", "0.4.11+commit.68ef5810"),
  // ...
];

impl SolcVersionFetcher {
  pub fn new() -> Self {
    let solc_long_versions: HashMap<String, String> = SOLC_VERSIONS.iter().map(|(s, l)| (s.to_string(), l.to_string())).collect();
    // ...
  }
}
```